crates/l2/Makefile

```
@@ -1,69 +1,70 @@
 
-down: down-local-l1 down-l2
+init: init-local-l1 contract-deps deploy-l1 init-l2 ## 🚀 Initializes a localnet with Lambda Ethereum Rust client as both L1 and L2
 
-clean: clean-local-l1 clean-contract-deps
+down: down-local-l1 down-l2 ## 🛑 Shuts down the localnet
 
-restart: restart-local-l1 restart-contract-deps restart-l2
+clean: clean-contract-deps ## 🧹 Cleans the localnet
 
-# Contracts
+restart: restart-local-l1 restart-contract-deps restart-l2 ## 🔄 Restarts the localnet
+
+cli: ## 🛠️ Builds the L2 Lambda Ethereum Rust CLI
+	cargo build --release --manifest-path ${ETHEREUM_RUST_PATH}/cmd/ethereum_rust/Cargo.toml
+
+# Variables
+
+ETHEREUM_RUST_PATH=$(shell pwd)/../../
+ETHEREUM_RUST_BIN_PATH=$(ETHEREUM_RUST_PATH)/target/release/ethereum_rust
+ETHEREUM_RUST_DEV_DOCKER_COMPOSE_PATH=$(ETHEREUM_RUST_PATH)/crates/blockchain/dev/docker-compose-dev.yaml
 
 FOUNDRY_PROJECT_HOME=$(shell pwd)/contracts
 L1_RPC_URL=http://localhost:8545
 L1_PRIVATE_KEY=0x385c546456b6a603a1cfcaa9ec9494ba4832da08dd6bcf4de9a71e4a01b74924
 
-contract-deps:
-	mkdir -p ${FOUNDRY_PROJECT_HOME}
-	forge install foundry-rs/forge-std --no-git --root ${FOUNDRY_PROJECT_HOME}
-	forge install succinctlabs/sp1-contracts --no-git --root ${FOUNDRY_PROJECT_HOME}
+# Local L1
 
-clean-contract-deps:
-	rm -rf contracts/lib
+init-local-l1: ## 🚀 Initializes an L1 Lambda Ethereum Rust Client
+	docker compose -f ${ETHEREUM_RUST_DEV_DOCKER_COMPOSE_PATH} up -d
 
-restart-contract-deps: clean-contract-deps contract-deps
+down-local-l1: ## 🛑 Shuts down the L1 Lambda Ethereum Rust Client
+	docker compose -f ${ETHEREUM_RUST_DEV_DOCKER_COMPOSE_PATH} down
 
-deploy-l1:
-	cd ${FOUNDRY_PROJECT_HOME} && \
-	forge script script/DeployL1.s.sol:DeployL1Script --rpc-url ${L1_RPC_URL} --private-key ${L1_PRIVATE_KEY} --broadcast
+restart-local-l1: down-local-l1 init-local-l1 ## 🔄 Restarts the L1 Lambda Ethereum Rust Client
 
-deploy-block-executor:
-	forge create ${FOUNDRY_PROJECT_HOME}/src/l1/OnChainOperator.sol:OnChainOperator --rpc-url ${L1_RPC_URL} --private-key ${L1_PRIVATE_KEY}
+# Contracts
 
-deploy-inbox:
-	forge create ${FOUNDRY_PROJECT_HOME}/src/l1/Inbox.sol:Inbox --rpc-url ${L1_RPC_URL} --private-key ${L1_PRIVATE_KEY}
+contract-deps: ## 📦 Installs the dependencies for the L1 contracts
+	mkdir -p ${FOUNDRY_PROJECT_HOME}
+	forge install foundry-rs/forge-std --no-git --root ${FOUNDRY_PROJECT_HOME} || exit 0
 
-# Local L1
+clean-contract-deps: ## 🧹 Cleans the dependencies for the L1 contracts.
+	rm -rf contracts/lib
 
-init-local-l1:
-	mkdir -p volumes/ volumes/reth volumes/reth/data
-	docker compose -f docker-compose-l2.yml up -d
+restart-contract-deps: clean-contract-deps contract-deps ## 🔄 Restarts the dependencies for the L1 contracts.
 
-down-local-l1:
-	docker compose -f docker-compose-l2.yml down
+deploy-l1: ## 📜 Deploys the L1 contracts
+	cd ${FOUNDRY_PROJECT_HOME} && \
+	forge script script/DeployL1.s.sol:DeployL1Script --rpc-url ${L1_RPC_URL} --private-key ${L1_PRIVATE_KEY} --broadcast
```

We can use the `solc` path here to set the path defined by the user.
```suggestion
	forge script script/DeployL1.s.sol:DeployL1Script --rpc-url ${L1_RPC_URL} --private-key ${L1_PRIVATE_KEY} --broadcast --use $$(which solc)
```